src/main.cpp

```diff
@@ -299,6 +299,20 @@ int main(int argc, char* argv[])
     bool nightly = true;
     boost::filesystem::path workPath = argv[0];
     workPath = workPath.parent_path();
+    // If the installation is the default one, update current installation
+#ifdef _WIN32
+    if(boost::filesystem::exists(workPath.parent_path() / "s25client.exe"))
+        workPath = workPath.parent_path();
+#elif defined(__APPLE__)
+    boost::filesystem::path tmpPath = workPath + "/../../../../../..";
+    if(boost::filesystem::exists(tmpPath / "s25client.app/Contents/MacOS/share/s25rttr/RTTR/s25update"))
+        workPath = tmpPath;
+#else
+    boost::filesystem::path tmpPath = workPath + "/../../..";
+    if(boost::filesystem::exists(tmpPath / "share/s25rttr/RTTR/s25update"))

```

Comment:
Well this is why the checks are done: If the path matches the expected path then take it, else take the updaters directory (as before) Or did I miss something? What is different when it is not the "standalone"-variant? Wouldn't this be caught by the check?